### ISSUE TITLE
Project `tags` cannot be specified when creating or updating a project

### ISSUE THREAD

**Issue by ctreatma** [equinix-labs/metal-go#126]
From https://github.com/equinix-labs/metal-go/pull/116

Starting with version `0.14.0`, metal-go announced support for `tags` on Metal projects.  However, the API spec only documents the presence of the `tags` property on projects that are received in API responses.  The spec does not include a way to provide `tags` when creating or updating a project.

**Comment by ctreatma** [comment]
It is possible to work around this limitation by using `AdditionalProperties` when creating or updating a project:

```go
projectCreateWithTags := metal.ProjectCreateFromRootInput{
  Name: "myProject",
  AdditionalProperties: map[string]interface{}{"tags": []string{"fooTag", "barTag"}},
}

resp, r, err := api_client.ProjectsApi.CreateProject(context.Background()).ProjectCreateFromRootInput(createInput).Execute()

projectUpdateWithTags := metal.ProjectUpdateInput{
  AdditionalProperties: map[string]interface{}{"tags": []string{"bazTag", "zogTag"}},
}

resp, r, err = api_client.ProjectsApi.UpdateProject(context.Background(), resp.GetId()).ProjectUpdateInput(updateInput).Execute()
```

**Comment by ctreatma** [comment]
Closed in `v0.18.0`
